Library/Homebrew/utils.rb

```diff
@@ -126,8 +126,8 @@ def self.install_gem_setup_path! gem, executable=gem
     ENV["PATH"] = "#{Gem.user_dir}/bin:#{ENV["PATH"]}"
 
     unless quiet_system "gem", "list", "--installed", gem
```

Comment:
It deliberately doesn't. My thinking was if someone wants to use their custom Ruby/Gem version, presuming it's the first Ruby/Gem in the `$PATH` and they have the required gem installed already, it'll work fine.

The only issue is when there's a custom Ruby/Gem prepending the `$PATH` but the required gems aren't installed, it won't install them in the correct location and can cause breakage. Essentially `/usr/bin/gem` is the fallback to the user's custom Ruby/Gem not having the required gems installed already.